api/data_explorer/controllers/export_url_controller.py

```diff
@@ -95,6 +109,25 @@ def _write_gcs_file(entities):
     current_app.logger.info(
         'Wrote gs://%s/%s' % (current_app.config['EXPORT_URL_GCS_BUCKET'],
                               random_str))
+    # Return in the format that signing a URL needs.
+    return '/' + current_app.config['EXPORT_URL_GCS_BUCKET'] + '/' + random_str
+
+
+def _create_signed_url(gcs_path):
+    creds = ServiceAccountCredentials.from_json_keyfile_name(PRIVATE_KEY_PATH)
+    service_account_email = current_app.config['DEPLOY_PROJECT_ID'] + '@appspot.gserviceaccount.com'
+    # Signed URL will be valid for 5 minutes
+    timestamp = str(int(time.time()) + 5 * 60)
+    str_to_sign = '\n'.join(['GET', '', '', timestamp, gcs_path])
```

Comment:
optional: I might name it file_metadata or something like that

Reply:
Done